lib/opam_build.ml

```diff
@@ -176,31 +175,53 @@ let install_project_deps ~opam_version ~opam_files ~selection =
 
 let spec ~base ~opam_version ~opam_files ~selection =
   let open Obuilder_spec in
+  let { Selection.variant; compatible_root_pkgs; _ } = selection in
   let to_name x = OpamPackage.of_string x |> OpamPackage.name_to_string in
   let home_dir =
     match Variant.os selection.Selection.variant with
     | `macOS -> "./src"
     | `linux -> "/src"
   in
-  let only_packages =
-    match selection.Selection.only_packages with
-    | [] -> ""
-    | pkgs -> " --only-packages=" ^ String.concat "," (List.map to_name pkgs)
+  let compatible_root_pkgs =
+    String.concat "," (List.map to_name compatible_root_pkgs)
   in
+  let compatible_root_pkgs_cmd = " --only-packages=" ^ compatible_root_pkgs in
+  let distro = Variant.distro variant in
   let run_build =
     match Variant.os selection.Selection.variant with
     | `macOS ->
         run
-          "cd ./src && opam exec -- dune build%s @install @check @runtest && \
-           rm -rf _build"
-          only_packages
+          {|cd ./src && opam exec -- dune build%s @install @check @runtest;
+res=$?;
+test "$res" != 1 && exit "$res";
+ret=1;
+for pkg in %s; do
+  if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"%s\""; then
+    echo -e "The package \033[1m$pkg\033[0m failed, but has been disabled for CI on \033[1m%s\033[0m using the \033[1mx-ci-accept-failures\033[0m field in its opam file.";
+    ret=0;
+  fi;
+done;
+exit "$ret";
```

Comment:
Instead of faking a job success, I think using the log matcher to display the skip is a much better way to go about it. See https://github.com/ocurrent/opam-repo-ci/blob/bca0f264ad1112310afef12f320b6291de08331d/service/main.ml#L61 for the log matcher rule to add.
```suggestion
for pkg in %s; do
  if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"%s\""; then
    echo "The package $pkg failed, but has been disabled for CI on %s using the x-ci-accept-failures field in its opam file.";
  fi;
done;
exit "$res";
```